synapse/lib/aha.py

```diff
@@ -1041,6 +1042,11 @@ async def getAhaSvcProxy(self, svcdef, timeout=None):
         if client is None:
             return None
 
+        try:
+            await asyncio.wait_for(client.ready.wait(), timeout=5.0)
+        except asyncio.TimeoutError:
+            return None
```

Comment:
This API was designed to raise asyncio.TimeoutError when a timeout condition occurs ( the proxy() method will raise it ) so this seems incorrect?

Reply:
The problem is using the default timeout and trying to wait() on a unreachable service. It never times out. /shrugs

Reply:
( if they want a timeout they must specify it. otherwise, who are we to disagree :D )